src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs

```diff
@@ -105,11 +111,130 @@ private void BuildCore(TocItemViewModel item, FileModel model, IHostService host
             }
         }
 
-        private static TocItemViewModel RestructureTableOfContent(TocItemViewModel toc, IHostService host)
+        private void RestructureCore(TocItemViewModel item, List<TocItemViewModel> parent, IList<TreeItemRestructure> restructures)
+        {
+            if (item.Items != null && item.Items.Count > 0)
+            {
+                var parentItems = new List<TocItemViewModel>(item.Items);
+                foreach (var i in item.Items)
+                {
+                    RestructureCore(i, parentItems, restructures);
+                }
+
+                item.Items = new TocViewModel(parentItems);
+            }
+
+            foreach (var restruction in restructures)
+            {
+                if (Matches(item, restruction))
+                {
+                    RestructureItem(item, parent, restruction);
+                }
+            }
+        }
+
+        private void RestructureItem(TocItemViewModel item, List<TocItemViewModel> parentItems, TreeItemRestructure restruction)
```

Comment:
also contains self... rename to items

---
In reply to: [97011268](https://github.com/dotnet/docfx/pull/1204#discussion_r97011268) [](ancestors = 97011268)